src/sql/connection.py

```diff
@@ -365,14 +369,55 @@ def _get_curr_connection_info(cls):
         }
 
     @classmethod
-    def _transpile_query(cls, query):
-        if not cls.current:
-            return query
+    def _get_curr_sqlglot_dialect(cls):
+        """Get the dialect name in sqlglot package scope
+
+        Returns:
+            str: Available dialect in sqlglot package, see more:
+            https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/dialect.py
+        """
         connection_info = cls._get_curr_connection_info()
+        if not connection_info:
+            return None
+
+        return DIALECT_NAME_SQLALCHEMY_TO_SQLGLOT_MAPPING.get(
+            connection_info["dialect"], connection_info["dialect"]
+        )
+
+    @classmethod
+    def _is_curr_dialect_support_backtick(cls):
+        """Get if the dialect support backtick (`) syntax as identifier
+
+        Raises:
+            Exception: Indicate there is unknown dialect support
+
+        Returns:
+            bool: If the dialect can use backtick identifier in the SQL clause
+        """
+        cur_dialect = cls._get_curr_sqlglot_dialect()
+        if not cur_dialect:
+            return False
         try:
-            write_dialect = DIALECT_NAME_SQLALCHEMY_TO_SQLGLOT_MAPPING.get(
-                connection_info["dialect"], connection_info["dialect"]
+            return (
+                "`" in sqlglot.Dialect.get_or_raise(cur_dialect).Tokenizer.IDENTIFIERS
             )
+        except Exception as e:
```

Comment:
1. We might still need to `except Exception as e`, to catch the original exception
2. We use `except Exception` because the `e` might be the three cases: 
  - ValueError, when dialect isn't defined in sqlglot
  - AttributeError, when dialect is defined, but no Tokenizer class declared inside
  - TypeError, when dialect && Tokenizer class are defined, but no IDENTIFIERS list declared inside


I rewrote in [Fix: not re-raise Exception in _is_curr_dialect_support_backtick](https://github.com/ploomber/jupysql/pull/318/commits/5100677b5c0f26349e3ca389ee59038c6640b755), but I think original way I did is wrong since it will re-raise new exception, also I added some test cases for this.

Please LMK if this you want 

Ref: [How do I raise the same Exception with a custom message in Python?](https://stackoverflow.com/questions/9157210/how-do-i-raise-the-same-exception-with-a-custom-message-in-python)

Reply:
Another approach is we handle those three exception cases in _is_curr_dialect_support_backtick directly, and return False when any of those happens

Implementation: [Maybe just return false?](https://github.com/ploomber/jupysql/pull/318/commits/9bd9aa1eee2fa98d09e54af2330ba95418414ced)

LMK which one you think is better?

Reply:
Because I think the purpose of `_is_curr_dialect_support_backtick` function is just return if the dialect can support backtick (`) syntax identifier, the caller function just need to know True / False is enough?

Since this function will treat all exception cases as not support, we don't have to handle the exception in the caller function since it's not support anyway